### PR TITLE
azure - Storage account permission issues exception messages +  Delete locked resource group should be logged and skipped

### LINKED ISSUE #4241: azure - storage account permission issues exception messages
We should handle permission issues when writing\reading Storage Queue\Blob to point to the appropriate access level for the SP.

### LINKED ISSUE #4341: azure - delete locked resource group should be logged and skipped
## Action
Try to delete a resource group that is locked

## Actual
Custodian is throwing an exception.

## Expected
Custodian should skip and log the error

```
(cloud-custodian) ➜  cloud-custodian git:(master) ✗ custodian-cask run -s=. policies/policy.yml
Custodian Cask (cloudcustodian/c7n:latest)
Skipped image pull - Last checked 23 minutes ago.

2019-07-08 20:57:46,374: custodian.azure.session:INFO Creating session with Azure CLI Authentication
2019-07-08 20:57:46,374: custodian.azure.session:INFO Session using Subscription ID: <my sub id>
2019-07-08 20:57:46,505: custodian.cache:WARNING Could not save cache /home/custodian/.cache/cloud-custodian.cache err: [Errno 2] No such file or directory: '/home/custodian/.cache/cloud-custodian.cache'
2019-07-08 20:57:46,506: custodian.cache:INFO Generating Cache directory: /home/custodian/.cache.
2019-07-08 20:57:49,283: custodian.policy:INFO policy:delete-empty-resource-groups resource:azure.resourcegroup region: count:1 time:2.90
2019-07-08 20:57:49,290: custodian.resources.resourcegroup:INFO Removing resource group test1
2019-07-08 20:57:49,383: custodian.output:ERROR Error while executing policy
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/c7n/policy.py", line 261, in run
    results = a.process(resources)
  File "/usr/local/lib/python3.7/site-packages/c7n_azure/resources/resourcegroup.py", line 93, in process
    self.manager.get_client().resource_groups.delete(group['name'])
  File "/usr/local/lib/python3.7/site-packages/azure/mgmt/resource/resources/v2018_05_01/operations/resource_groups_operations.py", line 231, in delete
    **operation_config
  File "/usr/local/lib/python3.7/site-packages/azure/mgmt/resource/resources/v2018_05_01/operations/resource_groups_operations.py", line 199, in _delete_initial
    raise exp
msrestazure.azure_exceptions.CloudError: Azure Error: ScopeLocked
Message: The scope '/subscriptions/<my sub id>/resourcegroups/test1' cannot perform delete operation because following scope(s) are locked: '/subscriptions/<my sub id>/resourceGroups/test1'. Please remove the lock and try again.
2019-07-08 20:57:49,384: custodian.commands:ERROR Error while executing policy delete-empty-resource-groups, continuing
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/c7n/commands.py", line 269, in run
    policy()
  File "/usr/local/lib/python3.7/site-packages/c7n/policy.py", line 945, in __call__
    resources = mode.run()
  File "/usr/local/lib/python3.7/site-packages/c7n/policy.py", line 261, in run
    results = a.process(resources)
  File "/usr/local/lib/python3.7/site-packages/c7n_azure/resources/resourcegroup.py", line 93, in process
    self.manager.get_client().resource_groups.delete(group['name'])
  File "/usr/local/lib/python3.7/site-packages/azure/mgmt/resource/resources/v2018_05_01/operations/resource_groups_operations.py", line 231, in delete
    **operation_config
  File "/usr/local/lib/python3.7/site-packages/azure/mgmt/resource/resources/v2018_05_01/operations/resource_groups_operations.py", line 199, in _delete_initial
    raise exp
msrestazure.azure_exceptions.CloudError: Azure Error: ScopeLocked
Message: The scope '/subscriptions/<my sub id>/resourcegroups/test1' cannot perform delete operation because following scope(s) are locked: '/subscriptions/<my sub id>/resourceGroups/test1'. Please remove the lock and try again.
```

### PR DESCRIPTION
**Pull request by anastasiia-zolochevska**
closes #4241
closes #4341

### CODE REVIEW

**Comment by stefangordon** [comment]
We don't usually test exception handlers unless they are doing something interesting, but as is typical with the codecov/patch gate, you'll have to invest in a test of minimal value.

It is an easy one, go find the `test_whitelist_not_authorized` test to see how to do it.

File: tools/c7n_azure/c7n_azure/actions/notify.py [hunk 1]
```diff
@@ -1,3 +1,4 @@
+from azure.common import AzureHttpError
```
Comment (aluong) [inline tools/c7n_azure/c7n_azure/actions/notify.py]: Can you add the Microsoft headers here?

File: tools/c7n_azure/c7n_azure/actions/notify.py [hunk 2]
```diff
@@ -89,5 +90,17 @@ def send_data_message(self, message, session):
...
-        return StorageUtilities.put_queue_message(queue_service, queue_name, self.pack(message)).id
+        try:
+            queue_service, queue_name = StorageUtilities.get_queue_client_by_uri(queue_uri, session)
+            return StorageUtilities.put_queue_message(
+                queue_service,
+                queue_name,
+                self.pack(message)).id
+        except AzureHttpError as e:
+            if e.status_code == 403:
+                self.log.error("Access error while putting message to the queue. "
```
Comment (aluong) [inline tools/c7n_azure/c7n_azure/actions/notify.py]: Alternative message: `Access Error: Storage Queue Data Contributor Role is required to enqueue messages.`

File: tools/c7n_azure/c7n_azure/output.py [hunk 3]
```diff
@@ -89,8 +89,13 @@ def upload(self):
                         blob_name,
                         os.path.join(root, f))
                 except AzureHttpError as e:
-                    self.log.error("Error writing output. Confirm output storage URL is correct "
-                                   "and that 'Storage Blob Contributor' role is assigned. \n" +
+                    if e.status_code == 403:
+                        self.log.error("Access error writing output. "
```
Comment (aluong) [inline tools/c7n_azure/c7n_azure/output.py]: I prefer if we do something like `Access Error: Storage Blob Data Contributor Role is required to write to Blob Storage.`

File: tools/c7n_azure/tests/test_output.py [hunk 4]
```diff
@@ -120,3 +123,47 @@ def test_app_insights_metrics(self):
...
+        gm.return_value = None, "logs", 'xyz'
+
+        output = self.get_azure_output()
+
+        output.blob_service = mock.MagicMock()
+        output.blob_service.create_blob_from_path = mock.MagicMock()
+        output.blob_service.create_blob_from_path.side_effect = AzureHttpError('not found', 404)
+
+        # Generate fake output file
+        with open(os.path.join(output.root_dir, "foo.txt"), "w") as fh:
```
Comment (logachev) [inline tools/c7n_azure/tests/test_output.py]: You can use any existing file in data folder.

File: tools/c7n_azure/tests/test_output.py [hunk 5]
```diff
@@ -120,3 +123,47 @@ def test_app_insights_metrics(self):
...
+            fh.write("abc")
+
+        output.upload()
+
+        args, _ = logger_mock.call_args
+
+        self.assertFalse(re.match("Access Error*", args[0]) is not None)
```
Comment (logachev) [inline tools/c7n_azure/tests/test_output.py]: nit: you can use `assertIsNone` and `assertIsNotNone`

**Pull request merged**